cmd/node/node.go

```diff
@@ -1200,7 +1200,7 @@ func (app *App) Start(ctx context.Context) error {
 	p2plog := app.addLogger(P2PLogger, lg)
 	// if addLogger won't add a level we will use a default 0 (info).
 	cfg.LogLevel = app.getLevel(P2PLogger)
-	app.host, err = p2p.New(ctx, p2plog, cfg, app.Config.Genesis.GenesisID(),
+	app.host, err = p2p.New(ctx, p2plog, cfg, app.Config.Genesis.GenesisID(), types.GetEffectiveGenesis()+1,
```

Comment:
i think it will be cleaner to pass only string instead of `app.Config.Genesis.GenesisID(), types.GetEffectiveGenesis()+1` to p2p.New

Reply:
Passing a string seems more general, if we ever wanted to run more than one host, the string could be useful.

Reply:
> You may want to run a host that doesn't have the genesis id or effective genesis in it's protocol prefix, so having a string would allow for that

i am curious about scenarios you are describing. can you give an example of why we want to run different hosts with different configuration?

Reply:
No I can't, I'm just thinking it makes for a more general interface, and I'm pretty bad at predicting the future, so I find general interfaces useful when things change in ways that I wasn't expecting.

Intuitively though having the p2p package (a networking package) know about genesis ids and layers, seems to be tying it into knowledge that is out of it's domain.